From e6c644683fbb34ac4ceb31150ef20a2536512149 Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Wed, 13 Jul 2022 20:08:47 +0100
Subject: [PATCH] Disable hardware acceleration

It seems to be broken with recent versions of WebKitGTK, especially on
X11. Needs further testing and possibly a bug report against WebKit
itself.

Fixes: #60
---
 devhelp/dh-web-view.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/devhelp/dh-web-view.c b/devhelp/dh-web-view.c
index dcd5b95a..89871da4 100644
--- a/devhelp/dh-web-view.c
+++ b/devhelp/dh-web-view.c
@@ -484,6 +484,7 @@ dh_web_view_constructed (GObject *object)
 
         webkit_settings = webkit_web_view_get_settings (WEBKIT_WEB_VIEW (view));
         webkit_settings_set_enable_back_forward_navigation_gestures (webkit_settings, TRUE);
+        webkit_settings_set_hardware_acceleration_policy (webkit_settings, WEBKIT_HARDWARE_ACCELERATION_POLICY_NEVER);
 
         /* Disable some things we have no need for. */
         webkit_settings_set_enable_html5_database (webkit_settings, FALSE);
-- 
GitLab

